action/update.go

```diff
@@ -73,6 +74,10 @@ func Update(installer *repo.Installer, skipRecursive, strip bool) {
 	// from the project. A removed dependency should warn and an added dependency
 	// should be added to the glide.yaml file. See issue #193.
 
+	if stripVendor {
+		confcopy = godep.RemoveGodepSubpackages(confcopy)
```

Comment:
codewise, do you want to hardcode anything godep in a non-godep dir?  If it were me, I would do something like:

main():
```
plugins := []ToolPlugin{NewGoToolPlugin()}
plugins = append(plugins, godep.ProbePlugins()...)
plugins = append(plugins, gom.ProbePlugins()...)
// etc
```

...where ToolPlugin is an interface.  One of the methods on ToolPlugin might be `RemoveVendoredJunk()`.  So here you would just loop over all plugins and have them remove their own vendored junk.

Something like that...